deepspeech.go

```diff
@@ -63,9 +63,51 @@ type sliceHeader struct {
 // buffer     A 16-bit, mono raw audio signal at the appropriate sample rate.
 // bufferSize The number of samples in the audio signal.
 // sampleRate The sample-rate of the audio signal.
-// TODO Make sure the C string is cleaned properly
 func (m *Model) SpeechToText(buffer []int16, bufferSize, sampleRate uint) string {
-	return C.GoString(C.STT(m.w, (*C.short)(unsafe.Pointer((*sliceHeader)(unsafe.Pointer(&buffer)).Data)), C.uint(bufferSize), C.uint(sampleRate)))
+	str := C.STT(m.w, (*C.short)(unsafe.Pointer((*sliceHeader)(unsafe.Pointer(&buffer)).Data)), C.uint(bufferSize), C.uint(sampleRate));
+	retval := C.GoString(str)
+	C.FreeString(str)
+	return retval
+}
+
+type MetadataItem C.struct_MetadataItem
+
+func (mi *MetadataItem) Character() string {
+	return C.GoString(C.GetCharacter((*C.MetadataItem)(unsafe.Pointer(mi))))
```

Comment:
@asticode Any way to define `GetCharacter` here rather than far away in another file?

Reply:
The other way is to add the C code directly at the top of this file but I don't think this is a good idea. FYI here's an example taken from the doc:

```go
package main

// #include <stdio.h>
// #include <stdlib.h>
//
// static void myprint(char* s) {
//   printf("%s\n", s);
// }
import "C"
import "unsafe"

func main() {
	cs := C.CString("Hello from stdio")
	C.myprint(cs)
	C.free(unsafe.Pointer(cs))
}
```